docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
I went back and looked. I see why its only an issue for 0.9.0.0. Its because a change went in that throws an exception for any version > 2 ([here](https://github.com/apache/kafka/blob/0.9.0.0/core/src/main/scala/kafka/cluster/Broker.scala#L60)), where in 0.8.x version isn't even used ([here](https://github.com/apache/kafka/blob/0.8/core/src/main/scala/kafka/cluster/Broker.scala#L36)). 

Can we leverage the fact that we are only adding a field, therefore this is a "compatible" update that does not require a version bump? That way we stay at version 2, 0.9.0.0 ignores the rack field, and everything works?

This should work because all of our json parsing just uses a Map[String, Any]. So adding fields doesn't break parsing. This PR actually acts as if version 2 & 3 are equal [here](https://github.com/allenxwang/kafka/blob/KAFKA-1215/core/src/main/scala/kafka/cluster/Broker.scala#L86-L93) anyway.